browser/src/map/Clipboard.js

```diff
@@ -929,22 +947,49 @@ L.Clipboard = L.Class.extend({
 
 	// Executes the navigator.clipboard.read() call, if it's available.
 	_navigatorClipboardRead: function(isSpecial) {
-		if (!L.Browser.clipboardApiAvailable) {
+		if (!L.Browser.clipboardApiAvailable && !window.ThisIsTheiOSApp) {
 			return false;
 		}
 
 		this._asyncAttemptNavigatorClipboardRead(isSpecial);
 		return true;
 	},
 
+	_iOSReadClipboard: async function() {
+		const encodedClipboardData = await window.webkit.messageHandlers.clipboard.postMessage('read');
+		const clipboardData = Array.from(
+			encodedClipboardData.split(' '),
+		).map((encoded) =>
+			(encoded === '(null)' ? '' : window.b64d(encoded)),
+		);
+
+		const dataByMimeType = {};
+
+		if (clipboardData[0]) {
+			dataByMimeType['text/plain'] = new Blob([clipboardData[0]]);
```

Comment:
are we sure it always appears in that order?    plain in index 0, html in index 1 ?

Reply:
Yep! The native implementation of this function is later down, and always returns in that order
```objc
            NSString * encodedPlainPayload = [plainPayload base64EncodedStringWithOptions:0];
            NSString * encodedHtmlPayload = [htmlPayload base64EncodedStringWithOptions:0];

            replyHandler([NSString stringWithFormat:@"%@ %@", encodedPlainPayload, encodedHtmlPayload], nil);
```